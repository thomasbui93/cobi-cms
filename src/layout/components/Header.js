/* @flow */
import React, { Component } from 'react';
import { withStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Button from '@material-ui/core/Button';
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';
import Typography from '@material-ui/core/Typography';
import PropTypes from 'prop-types';
import { TITLE, LOGOUT } from '../../services/constants';
import { connect } from 'react-redux';
import { toggleSideBar } from './../actions/layout';

const styles = {
  root: {
    flexGrow: 1,
  },
  flex: {
    flex: 1,
  },
  menuButton: {
    marginLeft: -12,
    marginRight: 20,
  },
};

export class Header extends Component {
  static propTypes = {
    openSideBar: PropTypes.func.isRequired,
    isMenuOpen: PropTypes.bool.isRequired,
    classes: PropTypes.object.isRequired
  }

  render() {
    const { classes } = this.props;
    return (
      <div className={classes.root}>
        <AppBar position="static">
          <Toolbar>
            <IconButton className={classes.menuButton}
              color="inherit"
              aria-label="Menu"
              onClick={this.props.openSideBar}>
              <MenuIcon />
            </IconButton>
            <Typography variant="title" color="inherit" className={classes.flex}>
              {TITLE}
            </Typography>
            <Button color="inherit"> {LOGOUT} </Button>
          </Toolbar>
        </AppBar>
      </div>
    )
  }
}

export const mapStateToProps = ({ layout }) => {
  return {
    isMenuOpen: layout.isMenuOpen
  }
}

export const mapDispatchToProps = (dispatch) => {
  return {
    openSideBar: () => {
      dispatch(toggleSideBar(true))
    }
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(withStyles(styles)(Header));