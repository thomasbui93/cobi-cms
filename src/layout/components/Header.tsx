import AppBar from '@material-ui/core/AppBar'
import Button from '@material-ui/core/Button'
import IconButton from '@material-ui/core/IconButton'
import Toolbar from '@material-ui/core/Toolbar'
import Typography from '@material-ui/core/Typography'
import MenuIcon from '@material-ui/icons/Menu'
import * as React from 'react'
import { connect } from 'react-redux'
import { Dispatch } from 'redux'

import { LOGOUT, TITLE } from '../../services/constants'
import { IRootState } from '../../types/RootState'
import { toggleSideBar } from '../actions/layout'

interface IHeaderProps {
    isMenuOpen: boolean,
    openSideBar: () => void
}

export class Header extends React.Component<IHeaderProps, any> {
    public render() {
        return (
            <div >
                <AppBar position="static">
                    <Toolbar>
                        <IconButton
                            color="inherit"
                            aria-label="Menu"
                            onClick={this.props.openSideBar}>
                            <MenuIcon />
                        </IconButton>
                        <Typography variant="title" color="inherit">
                            {TITLE}
                        </Typography>
                        <Button color="inherit"> {LOGOUT} </Button>
                    </Toolbar>
                </AppBar>
            </div>
        )
    }
}

export const mapStateToProps = (state: IRootState) => {
    return {
        isMenuOpen: state.layout.get('isMenuOpen')
    }
}

export const mapDispatchToProps = (dispatch: Dispatch) => {
    return {
        openSideBar: () => {
            dispatch(toggleSideBar(true))
        }
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(Header);
