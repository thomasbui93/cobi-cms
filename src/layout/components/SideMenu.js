/* @flow */

import React, { Component } from 'react'
import MenuList from '@material-ui/core/MenuList'
import MenuItem from '@material-ui/core/MenuItem'
import ListItemIcon from '@material-ui/core/ListItemIcon'
import ListItemText from '@material-ui/core/ListItemText'
import InboxIcon from '@material-ui/icons/MoveToInbox'
import DraftsIcon from '@material-ui/icons/Drafts'
import SendIcon from '@material-ui/icons/Send'
import SwipeableDrawer from '@material-ui/core/SwipeableDrawer'
import IconButton from '@material-ui/core/IconButton'
import ChevronLeftIcon from '@material-ui/icons/ChevronLeft'
import Divider from '@material-ui/core/Divider'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { toggleSideBar } from './../actions/layout'

export class SideMenu extends Component {
  static propTypes = {
    isMenuOpen: PropTypes.bool.isRequired,
    hideSideBar: PropTypes.func.isRequired,
    openSideBar: PropTypes.func.isRequired,
  }
  render() {
    return (
      <SwipeableDrawer
        open={this.props.isMenuOpen}
        onClose={this.props.hideSideBar}
        onOpen={this.props.openSideBar}
        >
        <div className="sidebar">
          <div>
            <IconButton onClick={this.props.hideSideBar}>
              <ChevronLeftIcon />
            </IconButton>
          </div>
          <Divider />
          <div className="sidebar__content">
            <MenuList>
              <MenuItem>
                <ListItemIcon >
                  <SendIcon />
                </ListItemIcon>
                <ListItemText inset primary="Sent mail" />
              </MenuItem>
              <MenuItem>
                <ListItemIcon>
                  <DraftsIcon />
                </ListItemIcon>
                <ListItemText inset primary="Drafts" />
              </MenuItem>
              <MenuItem>
                <ListItemIcon>
                  <InboxIcon />
                </ListItemIcon>
                <ListItemText inset primary="Inbox" />
              </MenuItem>
            </MenuList>
          </div>
        </div>
      </SwipeableDrawer>
    )
  }
}

const mapStateToProps = ({ layout }) => {
  return {
    isMenuOpen: layout.get('isMenuOpen')
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    hideSideBar: () => {
      dispatch(toggleSideBar(false))
    },
    openSideBar: () => {
      dispatch(toggleSideBar(true))
    }
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(SideMenu)