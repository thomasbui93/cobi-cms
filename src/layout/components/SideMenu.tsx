import Divider from '@material-ui/core/Divider'
import IconButton from '@material-ui/core/IconButton'
import ListItemIcon from '@material-ui/core/ListItemIcon'
import ListItemText from '@material-ui/core/ListItemText'
import MenuItem from '@material-ui/core/MenuItem'
import MenuList from '@material-ui/core/MenuList'
import SwipeableDrawer from '@material-ui/core/SwipeableDrawer'
import ChevronLeftIcon from '@material-ui/icons/ChevronLeft'
import DraftsIcon from '@material-ui/icons/Drafts'
import InboxIcon from '@material-ui/icons/MoveToInbox'
import SendIcon from '@material-ui/icons/Send'
import * as React from 'react'
import { connect } from 'react-redux'
import { Dispatch } from 'redux'

import { IRootState } from '../../types/RootState'
import { toggleSideBar } from '../actions/layout'


interface ISideMenuProps {
    isMenuOpen: boolean,
    hideSideBar: () => void,
    openSideBar: () => void
}

export class SideMenu extends React.Component<ISideMenuProps> {
    public render() {
        return (
            <SwipeableDrawer
                open={this.props.isMenuOpen}
                onClose={this.props.hideSideBar}
                onOpen={this.props.openSideBar}
            >
                <div className="sidebar">
                    <div>
                        <IconButton onClick={this.props.hideSideBar}>
                            <ChevronLeftIcon />
                        </IconButton>
                    </div>
                    <Divider />
                    <div className="sidebar__content">
                        <MenuList>
                            <MenuItem>
                                <ListItemIcon >
                                    <SendIcon />
                                </ListItemIcon>
                                <ListItemText inset={true} primary="Sent mail" />
                            </MenuItem>
                            <MenuItem>
                                <ListItemIcon>
                                    <DraftsIcon />
                                </ListItemIcon>
                                <ListItemText inset={true} primary="Drafts" />
                            </MenuItem>
                            <MenuItem>
                                <ListItemIcon>
                                    <InboxIcon />
                                </ListItemIcon>
                                <ListItemText inset={true} primary="Inbox" />
                            </MenuItem>
                        </MenuList>
                    </div>
                </div>
            </SwipeableDrawer>
        )
    }
}

export const mapStateToProps = (state: IRootState) => {
    return {
        isMenuOpen: state.layout.get('isMenuOpen')
    }
}

const mapDispatchToProps = (dispatch: Dispatch) => {
    return {
        hideSideBar: () => {
            dispatch(toggleSideBar(false))
        },
        openSideBar: () => {
            dispatch(toggleSideBar(true))
        }
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(SideMenu)