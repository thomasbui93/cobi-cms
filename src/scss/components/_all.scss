@import "header/global";
@import "header/search-box";