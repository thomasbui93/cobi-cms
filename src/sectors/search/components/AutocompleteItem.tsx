import * as React from 'react'
import { Link } from 'react-router-dom'
import { InterfaceSearchBoxAutocompleteItem } from '../reducers/state'

export class AutocompleteItem extends React.Component<InterfaceSearchBoxAutocompleteItem, any> {
    public render() {
        return (
            <Link className='search-box__item' to={`/content/${this.props.item.id}`}>
                { 
                    this.props.item.thumbnail ? 
                    <div className="search-box__thumbnail">
                        <img src={this.props.item.thumbnail} alt={this.props.item.title}/>
                    </div> : '' 
                }
                <div className='search-box__info'>
                    <div className='search-box__title'>{this.props.item.title}</div>
                    <div className='search-box__description'>{this.props.item.description}</div>
                </div>
            </Link>
        )
    }
}
