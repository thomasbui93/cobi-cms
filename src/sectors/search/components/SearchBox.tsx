import * as React from 'react'
import 'whatwg-fetch'
import { connect } from 'react-redux'
import { Dispatch } from 'redux'
import { Link } from 'react-router-dom'

import { AutocompleteItem } from './AutocompleteItem'
import { InterfaceSearchBoxProps, InterfaceSearchBoxState } from '../reducers/state'
import { requestAutocomplete } from '../actions/autocomplete'

interface InterfaceSearchBoxComponentProps extends InterfaceSearchBoxProps {
    requestAutocomplete: (queryValue: string) => any
}

export class SearchBox extends React.Component<InterfaceSearchBoxComponentProps, InterfaceSearchBoxState> {
    public static defaultProps: Partial<InterfaceSearchBoxProps> = {
        items: [],
        isLoading: false,
        error: false
    }

    public state: InterfaceSearchBoxState = {
        queryValue: '',
        displayAutoCompleteBox: false
    }

    public constructor(props: InterfaceSearchBoxComponentProps) {
        super(props)
        this.queryAutocomplete = this.queryAutocomplete.bind(this)
        this.isEnableAutoComplete = this.isEnableAutoComplete.bind(this)
        this.handleEscapeInteractions = this.handleEscapeInteractions.bind(this)
    }

    public queryAutocomplete(event: any) {
        const queryValue = event.target.value
        this.setState({ queryValue })
        if (this.isEnableAutoComplete()) {
            this.props.requestAutocomplete(queryValue)
        }
    }

    public isEnableAutoComplete():boolean {
        return this.state.queryValue.length >= 3 && this.state.displayAutoCompleteBox;
    }

    public handleEscapeInteractions(event: any) {
        if(event.keyCode === 27 || typeof event.keyCode === 'undefined') {
            this.setState({
                displayAutoCompleteBox: false
            })
        } else {
            this.setState({
                displayAutoCompleteBox: true
            })
        }
    }

    public render() {
        return (
            <div className='search-box'>
                <div className='search-box__input'>
                    <input type='text'
                        value={this.state.queryValue}
                        onKeyDown={this.handleEscapeInteractions}
                        onBlur={this.handleEscapeInteractions}
                        onChange={this.queryAutocomplete}
                        placeholder='Enter your query' />
                </div>
                {
                    this.isEnableAutoComplete() ?
                    <div className='search-box__autocomplete'>
                        {
                            this.props.isLoading ?
                                <span>Loading...</span> :
                                this.props.items.map(item => <AutocompleteItem item={item} key={item.id} />)
                        }
                        {
                            this.state.queryValue.length !== 0 ?
                                <Link to={`/search/${this.state.queryValue}`} className='button is-primary search-box__submit'>
                                    Search all content including {this.state.queryValue}
                                </Link> : ''
                        }
                    </div> : ''
                }
                
            </div>
        )
    }
}

export function mapStateToProps({ autocomplete }: { autocomplete: InterfaceSearchBoxProps }) {
    return {
        items: autocomplete.items,
        error: autocomplete.error,
        isLoading: autocomplete.isLoading
    }
}

export function mapDispatchToProps(dispatch: Dispatch) {
    return {
        requestAutocomplete: (queryValue: string) => dispatch(requestAutocomplete(queryValue))
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(SearchBox)