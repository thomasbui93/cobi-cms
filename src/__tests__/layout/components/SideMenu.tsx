import { shallow } from 'enzyme'
import { Map } from 'immutable'
import * as React from 'react'
import { mapStateToProps, SideMenu } from '../../../layout/components/SideMenu'

describe('mapStateToProps', () => {
    it('should map state to isMenuOpen value', () => {
        const state = {
            layout: Map({ isMenuOpen: true})
        }
        const props = mapStateToProps(state);
        expect(props).toEqual({ isMenuOpen: true})
    })
})

describe('Header component', () => {
    it('should render component without crashing', () => {
        const openSideBar = () => {
            return;
        };
        const hideSideBar = () => {
            return;
        };
        const wrapper = shallow(<SideMenu openSideBar={openSideBar} isMenuOpen={true} hideSideBar={hideSideBar}/>);
        expect(wrapper).toMatchSnapshot();
    })
})