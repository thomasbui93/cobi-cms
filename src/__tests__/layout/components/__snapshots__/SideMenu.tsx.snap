// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render component without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SideMenu
    hideSideBar={[Function]}
    isMenuOpen={true}
    openSideBar={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="sidebar"
      >
        <div>
          <WithStyles(IconButton)
            onClick={[Function]}
          >
            <pure(ChevronLeft) />
          </WithStyles(IconButton)>
        </div>
        <WithStyles(Divider) />
        <div
          className="sidebar__content"
        >
          <MenuList>
            <WithStyles(MenuItem)>
              <WithStyles(ListItemIcon)>
                <pure(Send) />
              </WithStyles(ListItemIcon)>
              <WithStyles(ListItemText)
                inset={true}
                primary="Sent mail"
              />
            </WithStyles(MenuItem)>
            <WithStyles(MenuItem)>
              <WithStyles(ListItemIcon)>
                <pure(Drafts) />
              </WithStyles(ListItemIcon)>
              <WithStyles(ListItemText)
                inset={true}
                primary="Drafts"
              />
            </WithStyles(MenuItem)>
            <WithStyles(MenuItem)>
              <WithStyles(ListItemIcon)>
                <pure(MoveToInbox) />
              </WithStyles(ListItemIcon)>
              <WithStyles(ListItemText)
                inset={true}
                primary="Inbox"
              />
            </WithStyles(MenuItem)>
          </MenuList>
        </div>
      </div>,
      "onClose": [Function],
      "onOpen": [Function],
      "open": true,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <WithStyles(IconButton)
              onClick={[Function]}
            >
              <pure(ChevronLeft) />
            </WithStyles(IconButton)>
          </div>,
          <WithStyles(Divider) />,
          <div
            className="sidebar__content"
          >
            <MenuList>
              <WithStyles(MenuItem)>
                <WithStyles(ListItemIcon)>
                  <pure(Send) />
                </WithStyles(ListItemIcon)>
                <WithStyles(ListItemText)
                  inset={true}
                  primary="Sent mail"
                />
              </WithStyles(MenuItem)>
              <WithStyles(MenuItem)>
                <WithStyles(ListItemIcon)>
                  <pure(Drafts) />
                </WithStyles(ListItemIcon)>
                <WithStyles(ListItemText)
                  inset={true}
                  primary="Drafts"
                />
              </WithStyles(MenuItem)>
              <WithStyles(MenuItem)>
                <WithStyles(ListItemIcon)>
                  <pure(MoveToInbox) />
                </WithStyles(ListItemIcon)>
                <WithStyles(ListItemText)
                  inset={true}
                  primary="Inbox"
                />
              </WithStyles(MenuItem)>
            </MenuList>
          </div>,
        ],
        "className": "sidebar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <WithStyles(IconButton)
              onClick={[Function]}
            >
              <pure(ChevronLeft) />
            </WithStyles(IconButton)>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <pure(ChevronLeft) />,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <MenuList>
              <WithStyles(MenuItem)>
                <WithStyles(ListItemIcon)>
                  <pure(Send) />
                </WithStyles(ListItemIcon)>
                <WithStyles(ListItemText)
                  inset={true}
                  primary="Sent mail"
                />
              </WithStyles(MenuItem)>
              <WithStyles(MenuItem)>
                <WithStyles(ListItemIcon)>
                  <pure(Drafts) />
                </WithStyles(ListItemIcon)>
                <WithStyles(ListItemText)
                  inset={true}
                  primary="Drafts"
                />
              </WithStyles(MenuItem)>
              <WithStyles(MenuItem)>
                <WithStyles(ListItemIcon)>
                  <pure(MoveToInbox) />
                </WithStyles(ListItemIcon)>
                <WithStyles(ListItemText)
                  inset={true}
                  primary="Inbox"
                />
              </WithStyles(MenuItem)>
            </MenuList>,
            "className": "sidebar__content",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <WithStyles(MenuItem)>
                  <WithStyles(ListItemIcon)>
                    <pure(Send) />
                  </WithStyles(ListItemIcon)>
                  <WithStyles(ListItemText)
                    inset={true}
                    primary="Sent mail"
                  />
                </WithStyles(MenuItem)>,
                <WithStyles(MenuItem)>
                  <WithStyles(ListItemIcon)>
                    <pure(Drafts) />
                  </WithStyles(ListItemIcon)>
                  <WithStyles(ListItemText)
                    inset={true}
                    primary="Drafts"
                  />
                </WithStyles(MenuItem)>,
                <WithStyles(MenuItem)>
                  <WithStyles(ListItemIcon)>
                    <pure(MoveToInbox) />
                  </WithStyles(ListItemIcon)>
                  <WithStyles(ListItemText)
                    inset={true}
                    primary="Inbox"
                  />
                </WithStyles(MenuItem)>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <WithStyles(ListItemIcon)>
                      <pure(Send) />
                    </WithStyles(ListItemIcon)>,
                    <WithStyles(ListItemText)
                      inset={true}
                      primary="Sent mail"
                    />,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": <pure(Send) />,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "inset": true,
                      "primary": "Sent mail",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <WithStyles(ListItemIcon)>
                      <pure(Drafts) />
                    </WithStyles(ListItemIcon)>,
                    <WithStyles(ListItemText)
                      inset={true}
                      primary="Drafts"
                    />,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": <pure(Drafts) />,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "inset": true,
                      "primary": "Drafts",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <WithStyles(ListItemIcon)>
                      <pure(MoveToInbox) />
                    </WithStyles(ListItemIcon)>,
                    <WithStyles(ListItemText)
                      inset={true}
                      primary="Inbox"
                    />,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": <pure(MoveToInbox) />,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "inset": true,
                      "primary": "Inbox",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="sidebar"
        >
          <div>
            <WithStyles(IconButton)
              onClick={[Function]}
            >
              <pure(ChevronLeft) />
            </WithStyles(IconButton)>
          </div>
          <WithStyles(Divider) />
          <div
            className="sidebar__content"
          >
            <MenuList>
              <WithStyles(MenuItem)>
                <WithStyles(ListItemIcon)>
                  <pure(Send) />
                </WithStyles(ListItemIcon)>
                <WithStyles(ListItemText)
                  inset={true}
                  primary="Sent mail"
                />
              </WithStyles(MenuItem)>
              <WithStyles(MenuItem)>
                <WithStyles(ListItemIcon)>
                  <pure(Drafts) />
                </WithStyles(ListItemIcon)>
                <WithStyles(ListItemText)
                  inset={true}
                  primary="Drafts"
                />
              </WithStyles(MenuItem)>
              <WithStyles(MenuItem)>
                <WithStyles(ListItemIcon)>
                  <pure(MoveToInbox) />
                </WithStyles(ListItemIcon)>
                <WithStyles(ListItemText)
                  inset={true}
                  primary="Inbox"
                />
              </WithStyles(MenuItem)>
            </MenuList>
          </div>
        </div>,
        "onClose": [Function],
        "onOpen": [Function],
        "open": true,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              <WithStyles(IconButton)
                onClick={[Function]}
              >
                <pure(ChevronLeft) />
              </WithStyles(IconButton)>
            </div>,
            <WithStyles(Divider) />,
            <div
              className="sidebar__content"
            >
              <MenuList>
                <WithStyles(MenuItem)>
                  <WithStyles(ListItemIcon)>
                    <pure(Send) />
                  </WithStyles(ListItemIcon)>
                  <WithStyles(ListItemText)
                    inset={true}
                    primary="Sent mail"
                  />
                </WithStyles(MenuItem)>
                <WithStyles(MenuItem)>
                  <WithStyles(ListItemIcon)>
                    <pure(Drafts) />
                  </WithStyles(ListItemIcon)>
                  <WithStyles(ListItemText)
                    inset={true}
                    primary="Drafts"
                  />
                </WithStyles(MenuItem)>
                <WithStyles(MenuItem)>
                  <WithStyles(ListItemIcon)>
                    <pure(MoveToInbox) />
                  </WithStyles(ListItemIcon)>
                  <WithStyles(ListItemText)
                    inset={true}
                    primary="Inbox"
                  />
                </WithStyles(MenuItem)>
              </MenuList>
            </div>,
          ],
          "className": "sidebar",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <WithStyles(IconButton)
                onClick={[Function]}
              >
                <pure(ChevronLeft) />
              </WithStyles(IconButton)>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <pure(ChevronLeft) />,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <MenuList>
                <WithStyles(MenuItem)>
                  <WithStyles(ListItemIcon)>
                    <pure(Send) />
                  </WithStyles(ListItemIcon)>
                  <WithStyles(ListItemText)
                    inset={true}
                    primary="Sent mail"
                  />
                </WithStyles(MenuItem)>
                <WithStyles(MenuItem)>
                  <WithStyles(ListItemIcon)>
                    <pure(Drafts) />
                  </WithStyles(ListItemIcon)>
                  <WithStyles(ListItemText)
                    inset={true}
                    primary="Drafts"
                  />
                </WithStyles(MenuItem)>
                <WithStyles(MenuItem)>
                  <WithStyles(ListItemIcon)>
                    <pure(MoveToInbox) />
                  </WithStyles(ListItemIcon)>
                  <WithStyles(ListItemText)
                    inset={true}
                    primary="Inbox"
                  />
                </WithStyles(MenuItem)>
              </MenuList>,
              "className": "sidebar__content",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <WithStyles(MenuItem)>
                    <WithStyles(ListItemIcon)>
                      <pure(Send) />
                    </WithStyles(ListItemIcon)>
                    <WithStyles(ListItemText)
                      inset={true}
                      primary="Sent mail"
                    />
                  </WithStyles(MenuItem)>,
                  <WithStyles(MenuItem)>
                    <WithStyles(ListItemIcon)>
                      <pure(Drafts) />
                    </WithStyles(ListItemIcon)>
                    <WithStyles(ListItemText)
                      inset={true}
                      primary="Drafts"
                    />
                  </WithStyles(MenuItem)>,
                  <WithStyles(MenuItem)>
                    <WithStyles(ListItemIcon)>
                      <pure(MoveToInbox) />
                    </WithStyles(ListItemIcon)>
                    <WithStyles(ListItemText)
                      inset={true}
                      primary="Inbox"
                    />
                  </WithStyles(MenuItem)>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      <WithStyles(ListItemIcon)>
                        <pure(Send) />
                      </WithStyles(ListItemIcon)>,
                      <WithStyles(ListItemText)
                        inset={true}
                        primary="Sent mail"
                      />,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": <pure(Send) />,
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {},
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "inset": true,
                        "primary": "Sent mail",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                  ],
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      <WithStyles(ListItemIcon)>
                        <pure(Drafts) />
                      </WithStyles(ListItemIcon)>,
                      <WithStyles(ListItemText)
                        inset={true}
                        primary="Drafts"
                      />,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": <pure(Drafts) />,
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {},
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "inset": true,
                        "primary": "Drafts",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                  ],
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      <WithStyles(ListItemIcon)>
                        <pure(MoveToInbox) />
                      </WithStyles(ListItemIcon)>,
                      <WithStyles(ListItemText)
                        inset={true}
                        primary="Inbox"
                      />,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": <pure(MoveToInbox) />,
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {},
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "inset": true,
                        "primary": "Inbox",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                  ],
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
